src/version.cpp

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (C) 2021 Emmanuel Engelhart <kelson@kiwix.org>
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation; either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * is provided AS IS, WITHOUT ANY WARRANTY; without even the implied
+ * warranty of MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE, and
+ * NON-INFRINGEMENT.  See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301 USA
+ *
+ */
+
+#include <iostream>
+
+#include <config.h>
+#include <zim/zim_config.h>
+#include <zstd.h>
+#include <lzma.h>
+#include <unicode/uvernum.h>
+
+#if defined(ENABLE_XAPIAN)
+#include <xapian.h>
+#endif
+
+namespace zim
+{
+
+void versions() {
+  std::cout << "libzim " << LIBZIM_VERSION << std::endl;
+  std::cout << "libzstd " << ZSTD_VERSION_STRING << std::endl;
+  std::cout << "liblzma " << LZMA_VERSION_STRING << std::endl;
+
+  // U_ICU_VERSION does not include the patch level if 0
+  std::cout << "libicu " << U_ICU_VERSION_MAJOR_NUM << "." <<
+    U_ICU_VERSION_MINOR_NUM << "." << U_ICU_VERSION_PATCHLEVEL_NUM << std::endl;
+
+#if defined(ENABLE_XAPIAN)
+  std::cout << "libxapian " << XAPIAN_VERSION << std::endl;
+#endif
+}
```

Comment:
I wonder if printing the versions is a good way to solve the problem being addressed. A better API may be to return the list of libraries and their versions. Then the final tool depending on libzim will enjoy the flexibility in formatting that information into end-user visible form and presenting it to the user (consider kiwix-serve reporting that information in HTML).

Reply:
Was my first intend as well, to return a `std::map<std::string, std::string>`. Thinking twice about it, I still think this is a good way to do, but actually I don't need really need this. A simple print to `stdout` is what I need (at least for the moment) and secures I don't duplicate the parsing of `std::map` and printing everywhere. It secures as well that the printing of dependences version is easy to update and the same output for all the tools. I don't say this is the best approach, but at least these are the rationals.

Reply:
Then the function better be renamed to `print_versions()`